tiledb/sm/cpp_api/query.h

```diff
@@ -382,7 +382,7 @@ class Query {
       auto attr_name = b_it.first;
       auto size_pair = b_it.second;
       auto var =
-          (attr_name != TILEDB_COORDS &&
+          (attr_name != TILEDB_COORDS && schema_.has_attribute(attr_name) &&
```

Comment:
Its also possible for string dimensions to be variable length, we might need to change the check to account for that too.